Restore correct coloring to selective callback

ogenstad opened the pull request:
This fixes the bug raised in #30506. While the bug exists in the `devel` branch it also exists in `stable-2.4`, so it would be great if this could be backported to 2.4.

##### SUMMARY
A bug was introduced in the selective callback plugin which caused the names of the colors to be printed instead of changing the actual output. This PR will restore the functionality while still allowing the user to configure which colors to use.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
- plugins/callback/selective
##### ANSIBLE VERSION
```
ansible 2.5.0 (coloring 8422a90561) last updated 2017/09/18 22:52:59 (GMT +200)
  config file = None
  configured module search path = [u'/Users/patrick/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/patrick/src/forks/ansible/lib/ansible
  executable location = /Users/patrick/src/forks/ansible/bin/ansible
  python version = 2.7.12 (default, Dec  5 2016, 11:55:02) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```
and:
```
ansible 2.4.0.0 (stable-2.4 3c805f85bc) last updated 2017/09/18 14:08:17 (GMT +200)
  config file = None
  configured module search path = [u'/Users/patrick/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/patrick/src/forks/ansible/lib/ansible
  executable location = /Users/patrick/src/forks/ansible/bin/ansible
  python version = 2.7.12 (default, Dec  5 2016, 11:55:02) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```
##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

ogenstad committed:
Restore correct coloring to selective callback

This fixes the bug raised in #30506

bcoca commented:
~~rerunning tests, failures don't seem related~~
change as instructed below

mattclay commented on lib/ansible/plugins/callback/selective.py:
```diff
@@ -41,18 +41,18 @@
 COLORS = {
     'normal': '\033[0m',
-    'ok': C.COLOR_OK,
+    'ok': '\033[{}m'.format(codeCodes[C.COLOR_OK]),
```
Zero length formats are not supported on python 2.6. Use this instead:

```python
'ok': '\033[{0}m'.format(codeCodes[C.COLOR_OK]),
```

ogenstad replied:
Ah, Python 2.6 :)

Looks like that's a general problem throughout this plugin. I can take a look at that.

ogenstad committed:
Fix format issues for Python 2.6 & indent

Removed the zero length fields to support format under Python 2.6

Fixed E128 continuation line under-indented for visual indent issue

ogenstad commented:
I've fixed the issues with '{}'.format(x) in Python 2.6 along with some old PEP8 indent issue. As mentioned earlier it would be nice if this could also be backported to Ansible 2.4.

bcoca commented:
backported, it will be part of 2.4.1
